relay-log/src/setup.rs

```diff
@@ -294,14 +295,25 @@ pub fn init(config: &LogConfig, sentry: &SentryConfig) {
         .init();
 
     if let Some(dsn) = sentry.enabled_dsn() {
+        let traces_sample_rate = config.traces_sample_rate;
+        // We're explicitly setting a `traces_sampler` here to circumvent trace
+        // propagation. A trace sampler that always just returns the constant
+        // `traces_sample_rate` is equivalent to using the `traces_sample_rate`
+        // directly, except it doesn't take into account whether the context
+        // was previously sampled. We don't want to take that into account because
+        // SDKs send headers with their envelopes that erroneously cause us to
+        // sample transactions.
```

Comment:
Please let me know if this comment doesn't explain the problem correctly or clearly enough.